### PR TITLE
cpu/stm32g4: add support + add nucleo-g474re board

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR adds support for the STM32 G4 family of CPUs and add support for the nucleo-g474re.

- What has been tested with success so far: i2c, spi, uart, rtc, rtt, timer, hwrng, flashpage
- ~What has been tested but doesn't fully work: flashpage. There's an issue with the raw mode.~
- What is configured but not tested in this port:
  - ~i2c: could be configured but doesn't work. Probably because of a I2C clock misconfiguration. I haven't looked in detail yet~
  -  ~spi~

The clock configuration uses a similar strategy as the one used in #14163. Depending on the one that is merged first, I'll factorize the 2.

### Testing procedure

- A green Murdock
- Run `./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . nucleo-g474re --jobs=4` and check the results are ok (will report later when done).

### Issues/PRs references

None

### CODE REVIEW

**Comment by aabadie** [comment]
Here are the results of the local tests:
```
Failures during test:
- [tests/driver_grove_ledbar](tests/driver_grove_ledbar/test.failed)
- [tests/driver_my9221](tests/driver_my9221/test.failed)
- [tests/mtd_flashpage](tests/mtd_flashpage/test.failed)
- [tests/periph_flashpage](tests/periph_flashpage/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
- [tests/pkg_libfixmath_unittests](tests/pkg_libfixmath_unittests/test.failed)
- [tests/unittests](tests/unittests/test.failed)
- [tests/xtimer_now_irq](tests/xtimer_now_irq/test.failed)
```

- `tests/driver_*`: I think there's an issue with openocd or something like this
- `tests/periph_timer_short_relative_set`: I think this one is known to fail anyway
- `tests/pkg_libfixmath_unittests`: no idea what is the issue
- `tests/mtd_flashpage` and `tests/periph_flashpage` are failing because the flashpage driver has an issue with the raw mode
- `tests/unittests` works but takes too much time to complete. Maybe the 170MHz (max) clock frequency should be the default
- `tests/xtimer_now_irq`: no idea of what could be the issue.

**Comment by aabadie** [comment]
`tests/periph_flashpage` and `tests/mtd_flashpage` are fixed by b6b1820.

**Comment by aabadie** [comment]
Regarding `tests/xtimer_now_irq`, the answer is in the README of the application: "This test does not work for 32 bit timers.". Which is the case here.

**Comment by aabadie** [comment]
Regarding `tests/pkg_libfixmath_unittests`, the test is failing on several other boards as well. I tested on p-nucleo-wb55 and stm32f429i-disc1 and got the same failure.

**Comment by aabadie** [comment]
Some updates in this PR: rtc and i2c are now functional. SPI is configured but I didn't test.

**Comment by benpicco** [comment]
Needs a rebase - but looks like everything is working now :smiley:

File: boards/common/stm32/include/g4/cfg_clock_default.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+#elif CLOCK_USE_HSE
+#define CLOCK_CORECLOCK     (CLOCK_HSE)
+
+#elif CLOCK_USE_PLL
+/* The following parameters configure a 80MHz system clock with HSE as input clock */
+#define CLOCK_PLL_M         (6)
+#define CLOCK_PLL_N         (40)
+#define CLOCK_PLL_R         (2)
+/* Use the following to reach 170MHz
```
Comment (benpicco) [inline boards/common/stm32/include/g4/cfg_clock_default.h:55]: Can we use 
```C
#if CLOCK_CORECLOCK >= 170000000
```

Comment (aabadie) [reply boards/common/stm32/include/g4/cfg_clock_default.h:55]: I tried and it works. But the manual says that the clock should be set to 80MHz during a short period before increasing it to 170MHz. This is not implemented in the clock.c file :)

**Comment by benpicco** [comment]
Want to add

```C
#define STM32_BOOTLOADER_ADDR   (0x1FFF0000)
```

to `include/periph/g4/periph_cpu.h` so reset to [bootloader](https://www.st.com/resource/en/application_note/cd00167594-stm32-microcontroller-system-memory-boot-mode-stmicroelectronics.pdf) works.

**Comment by aabadie** [comment]
rebased and added the missing bootloader define.

**Comment by aabadie** [comment]
Initially I wanted to have #14152 merged first but it seems to take more time to be merged than I thought. If we want to have stm32g4 in the next release, then there's no need to wait.

**Comment by bergzand** [comment]
`examples/default` works on my nucleo-g474re :smiley:

**Comment by aabadie** [comment]
@benpicco, it's all green :)

**Pull request merged**